hedera-state-validator/src/main/java/com/hedera/statevalidation/parameterresolver/InitUtils.java

```diff
@@ -328,25 +254,20 @@ static ServicesRegistryImpl initServiceRegistry() {
      * This method initializes the State API
      */
     static void initServiceMigrator(State state, Configuration configuration, ServicesRegistry servicesRegistry) {
-        final var bootstrapConfigProvider = new BootstrapConfigProviderImpl();
         final var serviceMigrator = new OrderedServiceMigrator();
         final var platformFacade = PlatformStateFacade.DEFAULT_PLATFORM_STATE_FACADE;
         final var deserializedVersion = platformFacade.creationSoftwareVersionOf(state);
-        final var version = getNodeStartupVersion(bootstrapConfigProvider.getConfiguration());
+        final var nodeStartupVersion = readVersion();
```

Comment:
The way `readVersion()` is implemented is idiosyncratic to longevity testing in `perf1` and is broken for generic testing: no `VERSION` file is saved instate by the consensus node (nor is generated at all).